src/reflect/value.go

```diff
@@ -217,7 +217,15 @@ func (v Value) CanAddr() bool {
 }
 
 func (v Value) Addr() Value {
-	panic("unimplemented: (reflect.Value).Addr()")
+	if !v.CanAddr() {
+		panic("reflect.Value.Addr of unaddressable value")
+	}
+
+	return Value{
+		typecode: pointerTo(v.typecode),
+		value:    unsafe.Pointer(&v.value),
+		flags:    v.flags,
```

Comment:
I believe you need to remove the indirect flag here:

```suggestion
		flags:    v.flags ^ valueFlagIndirect,
```

because the following code:

```go
package main

import "reflect"

func main() {
        n := 5
        v := reflect.ValueOf(&n)
        println("CanAddr:", v.Elem().Addr().CanAddr())
}
```

Has a different output in TinyGo:

```
$ go run ./tmp/reflect.go
CanAddr: false

$ tinygo run ./tmp/reflect.go
CanAddr: true
```